src/graph/graphcontext.c

```diff
@@ -0,0 +1,54 @@
+#include "graphcontext.h"
+#include "graphcontext_type.h"
+
+GraphContext* GraphContext_New(RedisModuleCtx *ctx, Graph *g, const char *graph_name) {
+  GraphContext *gc = malloc(sizeof(GraphContext));
+  gc->g = g;
+  gc->graph_name = strdup(graph_name);
+  gc->label_strings = malloc(sizeof(char*) * gc->g->label_cap);
+
+  // TODO I will probably make a macro for building context keys on the stack
+  int keylen = strlen(graph_name) + 4;
+  char strKey[keylen + 1];
+  strKey[0] = '\0';
+  strcat(strKey, graph_name);
+  strcat(strKey, "_ctx");
+
+  RedisModuleString *rm_graphctx = RedisModule_CreateString(ctx, strKey, keylen);
+  RedisModuleKey *key = RedisModule_OpenKey(ctx, rm_graphctx, REDISMODULE_WRITE);
+  assert(RedisModule_ModuleTypeGetType(key) == REDISMODULE_KEYTYPE_EMPTY);
+  RedisModule_ModuleTypeSetValue(key, GraphContextRedisModuleType, g);
+  RedisModule_CloseKey(key);
+
+  return gc;
+}
+
+GraphContext *GraphContext_Get(RedisModuleCtx *ctx, const char *graph_name) {
+  GraphContext *gc = NULL;
+
+  int keylen = strlen(graph_name) + 4;
+  char strKey[keylen + 1];
+  strKey[0] = '\0';
+  strcat(strKey, graph_name);
+  strcat(strKey, "_ctx");
+
+  RedisModuleString *rm_graphctx = RedisModule_CreateString(ctx, strKey, keylen);
+  RedisModuleKey *key = RedisModule_OpenKey(ctx, rm_graphctx, REDISMODULE_WRITE);
+  if (RedisModule_ModuleTypeGetType(key) == GraphContextRedisModuleType) {
+    gc = RedisModule_ModuleTypeGetValue(key);
+  }
+
+  RedisModule_CloseKey(key);
+  return gc;
+}
+
+const char* GraphContext_GetLabel(const GraphContext *gc, int label_id) {
+  assert(gc && label_id < gc->g->label_count);
```

Comment:
I incline toward moving `label_count` and similar `Graph` members to the context.